feat(sdk-docs): add documentation configuration for SDK monorepo

jeffersonBastos opened the pull request:
# Description
This PR refactors the SDK section in the documentation to reflect the new monorepo structure.
It also introduces three new tabs to organize the packages as follows:

├── cow-sdk (umbrella)
│
|── Provider Adapters
│   ├── cow-ethers-v5-adapter
│   ├── cow-ethers-v6-adapter
│   └── cow-viem-adapter
│
|── Core Utilities
│   ├── cow-common
│   ├── cow-config
│   └── cow-contracts-ts
│
|── Protocol Components
│   ├── sdk-order-signing
│   ├── sdk-order-book
│   ├── sdk-app-data
│   ├── sdk-trading
│   ├── sdk-composable
│   ├── sdk-shed
│   └── sdk-bridging
│
└── app-data
Note:
This PR is still referencing the main branch. Because of that, the full SDK feature branch must be merged into main before this PR can be finalized.

https://github.com/cowprotocol/cow-sdk/tree/SDK-AGNOSTIC-LIB

# Changes

The documentation structure was refactored to include three separate sections for individual package documentation. This change was required because TypeDoc's type resolution system cannot properly handle cross-package references when documenting only the umbrella package with aggregated exports.

By documenting each package independently, we ensure:
- Proper type resolution across package boundaries
- Accurate API documentation with correct type references
- Better developer experience with clear package separation

## Related Issues

https://github.com/cowprotocol/cow-sdk/pull/403


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Fixed note markdown rendering, added CoW AMM Deployer tutorial, and reorganized SDK docs with new categories and improved sidebars.
  * Expanded automated API/docs generation coverage for additional SDK components and adapters.
  * Updated contributing, README, and deployment guidance; noted Vercel deployment.

* **Chores**
  * Adjusted docs ignore rules, simplified external build scripts and standardized on pnpm, and increased Node.js memory for site builds.
  * Added build command for deployment config.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yvesfracari committed:
Add CoW AMM Deployer tutorial

yvesfracari committed:
Apply mfw78 suggestions

Co-authored-by: mfw78 <53399572+mfw78@users.noreply.github.com>

yvesfracari committed:
fix price oracle wrong uniswap version

Co-authored-by: Fábio Mendes <43360747+mendesfabio@users.noreply.github.com>

jeffersonBastos committed:
Merge branch 'cowprotocol:main' into main

jeffersonBastos committed:
feat(sdk-docs): add documentation configuration for SDK monorepo

jeffersonBastos commented:
I have read the CLA Document and I hereby sign the CLA

alfetopito commented on docs/cow-protocol/reference/sdks/.gitignore:
```diff
@@ -1,3 +1,10 @@
 **/*
 !_category_.json
+!_category_.yml
```
`!_category_.json` rule is already in the file

alfetopito commented on docusaurus.config.ts:
```diff
@@ -89,20 +91,260 @@ const config: Config = {
+        id: 'sdk-ethers-v5-adapter',
+        // TypeDoc options
+        entryPoints: ['./external/cow-sdk/packages/providers/ether-v5-adapter/src/index.ts'],
+        tsconfig: './external/cow-sdk/packages/providers/ether-v5-adapter/tsconfig.json',
```
@jeffersonBastos shouldn't it be `ethers` instead?

jeffersonBastos replied:
@alfetopito  Yes, it should all be "ethers" actually.
The one without the "s" is the folder name in the SDK. Would you like me to make that change?

shoom3301 replied:
I will do that

shoom3301 replied:
@jeffersonBastos I renamed it in cow-sdk repo, please update it in this PR

alfetopito commented on package.json:
```diff
@@ -6,11 +6,9 @@
-    "build:external:appdata": "yarn git:clone:appdata && cd external/app-data && yarn install --frozen --ignore-scripts && yarn compile",
-    "build:external": "yarn build:external:cowsdk && yarn build:external:appdata",
-    "build": "yarn build:external && docusaurus build",
+    "build:external:cowsdk": "cd external/cow-sdk && pnpm install && pnpm run build",
```
Good point.

@jeffersonBastos if pnpm is now a requirement for the build, let's add it to the dependencies

jeffersonBastos replied:
Nice catch!
 - I've added the buildCommand: "npm install -g pnpm && yarn build" to vercel.json to ensure pnpm is available during Vercel deployments, and updated the documentation to instruct developers to install pnpm locally.
However, I noticed that this project doesn't have any GitHub Actions CI workflows that run builds - the existing workflows only handle CLA, project board management, and Slack notifications. The project appears to rely entirely on Vercel for deployment.

jeffersonBastos committed:
chore(docs): fix pnpm dependency for build process

- Add pnpm installation to Vercel build command
- Update documentation to include pnpm requirement

jeffersonBastos committed:
chore(docs): change ether to ethers SDK named path

jeffersonBastos committed:
Merge branch 'main' into feat/sdk-agnostic-docs

jeffersonBastos committed:
Merge branch 'main' into feat/sdk-agnostic-docs
